module/modmanager/manager.go

```diff
@@ -121,6 +135,83 @@ func (mgr *Manager) Add(ctx context.Context, cfg config.Module) error {
 	return nil
 }
 
+// Reconfigure reconfigures an existing resource module and returns the names of
+// now orphaned resources.
```

Comment:
Good idea to actually handle orphans! But I think there's a lot of nuances to work through on this.